### PR TITLE
feat(migrate): add cleanup-oauth action for workbench migration

### PR DESCRIPTION
**Pull request by chambridge**
## Description
Implement the cleanup-oauth action that removes legacy OAuth-proxy resources (Route, Service, Secrets, OAuthClient) left behind after migrating workbenches from RHOAI 2.x to 3.x.

- Add cleanup action in pkg/migrate/actions/workbenches/cleanup/ with support for dry-run mode and per-resource error handling
- Extract SharedScopeOptions from the kueue action into pkg/migrate/actions/workbenches/options.go to eliminate duplicate notebook listing and validation logic between actions
- Refactor kueue action to use the shared scope options
- Add OAuthClient resource type to pkg/resources/types.go
- Register cleanup-oauth action in the migrate registry
- Add comprehensive test suite with 36 test cases covering all code paths including error scenarios and edge cases

Relates-to: [RHOAIENG-72119](https://redhat.atlassian.net/browse/RHOAIENG-72119)

## Testing

- [ ] Unit tests pass (`make test`)
- [ ] Linter passes (`make lint`)
- [ ] New functionality includes tests

## Review checklist

- [ ] Code follows project conventions (see docs/coding/)
- [ ] Changes are documented if needed

## Summary by CodeRabbit

* **New Features**
  * Added a new cleanup action to remove legacy OAuth-related resources after notebook migration.
  * Added shared workbench selection options, including filtering by notebook name or namespace.
  * Updated existing migration actions to use the same selection behavior.

* **Bug Fixes**
  * Improved handling of already-migrated or missing resources so cleanup runs safely and idempotently.
  * Added clearer confirmation and dry-run behavior during migration cleanup.

### CODE REVIEW

File: pkg/migrate/actions/workbenches/cleanup/cleanup.go [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+	if hasOAuthProxy {
+		failures = append(failures, "legacy oauth-proxy sidecar still present")
+	}
+
+	if injectOAuth, exists := annotations[annotationInjectOAuth]; exists {
+		if !hasKubeRBACProxy || hasOAuthProxy {
+			failures = append(failures,
+				fmt.Sprintf("legacy inject-oauth annotation still exists: %q", injectOAuth))
+		}
+	}
```
[the remark this answers is not in the document]
Comment (chambridge) [reply pkg/migrate/actions/workbenches/cleanup/cleanup.go:132]: The conditional tolerance is intentional and matches the shell script exactly (L714-730). When kube-rbac-proxy is present and `oauth-proxy` is absent, the notebook is migrated — the lingering annotation is harmless. Unconditionally failing would produce false negatives on properly migrated notebooks.

File: pkg/migrate/actions/workbenches/cleanup/cleanup.go [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+
+func (a *CleanupOAuthAction) promptCleanupContinueOrSkip(
+	target action.Target,
+	name string,
+	namespace string,
+	failures []string,
+) bool {
+	if target.SkipConfirm {
+		return true
+	}
```
[the remark this answers is not in the document]
Comment (chambridge) [reply pkg/migrate/actions/workbenches/cleanup/cleanup.go:235]: The shell script does the same thing (L318-323): `--yes` means "answer all prompts affirmatively." Pre-check failures are still recorded in step output for visibility. This is the expected contract for automation.

File: pkg/migrate/actions/workbenches/cleanup/cleanup.go [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+		step.Recordf(stepID, "Failed to check %s: %v", result.StepFailed, label, err)
+
+		return false
+	}
+
+	if namespace != "" {
+		err = ri.Namespace(namespace).Delete(ctx, name, metav1.DeleteOptions{})
+	} else {
+		err = ri.Delete(ctx, name, metav1.DeleteOptions{})
+	}
```
[the remark this answers is not in the document]
Comment (chambridge) [reply pkg/migrate/actions/workbenches/cleanup/cleanup.go:200]: should be fixed in latest code.

**Review by robotmaxtron — APPROVED** [review]
/lgtm

**Review by carlkyrillos — APPROVED** [review]
